common/src/main/java/tachyon/retry/CountingRetry.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.retry;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * An option which allows retrying based on maximum count.
+ */
+public class CountingRetry implements RetryPolicy {
+
+  private final int mMaxRetries;
+  private int mCount = -1;
```

Comment:
The contract for `getRetryCount()` is that it returns the number of retries which have been done, so it should return `mCount`, where `mCount` starts at `0` and increases by `1` whenever a retry is attempted. I don't understand why we need to subtract `1` when `mCount` is nonzero.

If something is attempted 3 times, only the 2nd and 3rd attempts are retries; the first attempt is just a regular try.